pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -195,38 +219,91 @@ func TestActuator(t *testing.T) {
 		name                    string
 		machine                 *clusterv1.Machine
 		error                   string
-		operation               func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine)
+		operation               func(client client.Client, actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine)
 		describeInstancesOutput *ec2.DescribeInstancesOutput
 		runInstancesErr         error
 		describeInstancesErr    error
 		terminateInstancesErr   error
 		lbErr                   error
 	}{
 		{
-			name:    "Update with success",
+			name:    "Create machine with success",
 			machine: machine,
-			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+			operation: func(objectClient client.Client, actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				createErr := actuator.Create(context.TODO(), cluster, machine)
+				machineStatus, err := getMachineStatus(objectClient, machine)
+				if err != nil {
+					t.Fatalf("Unable to get machine status: %v", err)
+				}
+
+				assert.NoError(t, createErr)
+				assert.Equal(t, machineStatus.Conditions[0].Reason, MachineCreationSucceeded)
+
+				// Get the machine
+				if exists, err := actuator.Exists(context.TODO(), cluster, machine); err != nil || !exists {
+					t.Errorf("Instance for %v does not exists: %v", strings.Join([]string{machine.Namespace, machine.Name}, "/"), err)
+				} else {
+					t.Logf("Instance for %v exists", strings.Join([]string{machine.Namespace, machine.Name}, "/"))
+				}
+
+				// Update a machine
+				if err := actuator.Update(context.TODO(), cluster, machine); err != nil {
+					t.Errorf("Unable to create instance for machine: %v", err)
+				}
+
+				// Get the machine
+				if exists, err := actuator.Exists(context.TODO(), cluster, machine); err != nil || !exists {
```

Comment:
Isn't there a case here where there's no error, but it also doesn't exist?

Reply:
The else branch is executed only when there is no error and the machine exist. See `err != nil || !exists` condition.